src/validation.cpp

```diff
@@ -5230,6 +5256,136 @@ bool ChainstateManager::PopulateAndValidateSnapshot(
     return true;
 }
 
+// Currently, this function holds cs_main for its duration, which could be for
+// multiple minutes due to the GetUTXOStats call. This hold is necessary
+// because we need to avoid advancing the background validation chainstate
+// farther than the snapshot base block - and this function is also invoked
+// from within ConnectTip, i.e. from within ActivateBestChain, so cs_main is
+// held anyway.
```

Comment:
Perhaps it isn't actually necessary to hold `cs_main` throughout this if I set `m_stop_use` immediately, and then ensure all chainstate accesses are going through `isUsable()` checks...